cmake/Settings.cmake

```diff
@@ -0,0 +1,73 @@
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2024 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See http://swift.org/LICENSE.txt for license information
+# See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+# Settings intended to be applied to every Swift target in this package.
+# Analogous to project-level build settings in an Xcode project.
+add_compile_options(
```

Comment:
The reason I put it here was because of `TestingMacros`. We build `TestingMacros` as an `ExternalProject` but these settings are meant to apply to both Swift targets.  I didn't want to duplicate them.

Reply:
I broke these out into more granular places.  Let me know if you like it better; happy to iterate.